exercises.py

```diff
@@ -1,13 +1,168 @@
+#  spam = [2,4,'hello', 6, 8]
+# For the following three questions, let’s say spam contains the list ['a', 'b', 'c', 'd'].
+# 3. What does spam[int(int('3' * 2) // 11)] evaluate to? 
+# 'd'
+# 4. What does spam[-1] evaluate to? 
+# 'd'
+# 5. What does spam[:2] evaluate to?
+#  ['a', 'b', 'c']
+# For the following three questions, let’s say bacon contains the list [3.14, 'cat', 11, 'cat', True].
+# 6. What does bacon.index('cat') evaluate to?
+#  1
+# 7. What does bacon.append(99) make the list value in bacon look like? 
+#  bacon=[3.14, 'cat', 11, 'cat', True, 99]
+# 8. What does bacon.remove('cat') make the list value in bacon look like? 
+#  bacon=[3.14, 11, 'cat', True]
+# 9. What are the operators for list concatenation and list replication? 
+#  The operators are + to combine values and * to replicate list.
+
+# 10. What is the difference between the append() and insert() list methods? 
+#  append() will add an item to the end of a list and insert() allows one to choose where exactly in a list an item will go.
+
+# 11. What are two ways to remove values from a list? 
+#  remove() and del()
+
+# 12. Name a few ways that list values are similar to string values. 
+#  Both are called items, can be store as variables, and have to be contained within parameteres []for list and ''or ""for a string value.
+
+# 13. What is the difference between lists and tuples? 
+#  List use [], while tiples use (). Tuplues can not be modified, appended or removed.
+# 14. How do you type the tuple value that has just the integer value 42 in it?
+#  int(tuple('42'))
+# 15. How can you get the tuple form of a list value? How can you get the list form of a tuple value? 
+#  tuple([]) and list (()).
+# 16. Variables that “contain” list values don’t actually contain lists directly. What do they contain instead? 
+#  They contain a refernce to the list.
+# 17. What is the difference between copy.copy() and copy.deepcopy()? 
+#  copy.copy() will make dulicate of a list or dictionary and copy.deepcopy( ) will make a copy of list within list.
+
+#  # Chapter 5
+# 1. What does the code for an empty dictionary look like?
+#  dictionary = {}
+# 2. What does a dictionary value with a key 'foo' and a value 42 look like?
+#  {'foo':42}
+# 3. What is the main difference between a dictionary and a list? 
+#  Dictionaries are not order while list are.
+# 4. What happens if you try to access spam['foo'] if spam is {'bar': 100}? 
+#  You would get a KeyError error message.
+# 5. If a dictionary is stored in spam, what is the difference between the expressions 'cat' in spam and 'cat' in spam.keys()? 
+#  'cat' in spam.keys() will return the key associated with 'cat'.
+# 6. If a dictionary is stored in spam, what is the difference between the expressions 'cat' in spam and 'cat' in spam.values()?
+#  'cat' in spam.value() will return the value associated with 'cat'.
```

Comment:
What you describe is what `spam['cat']` will do. What does `spam.values()' return? 

`'cat' in <data structure>` asks if the string 'cat' is present in the data structure.